### PR TITLE
Don't import plugins that don't support android in settings.gradle

### PR DESCRIPTION
**Pull request by blasten**
## Description

Currently, the Flutter Gradle plugin skips plugins that don't have an `android/build.gradle` file. However, we shouldn't be importing these plugins in the first place. A new behavior in Android Studio or Gradle could break the current check. 

Android Studio currently creates an android/ noop directory in the `.pub-cache` directory.

Hopefully, when https://github.com/flutter/flutter/issues/40460 is fixed, we will have a tool to automate the migration of project files like `settings.gradle`.

## Related Issues

https://github.com/flutter/flutter/issues/39657

## Tests

Updated tests.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by jonahwilliams** [comment]
If we're making an edit to the settings.gradle, I think the right fix would be to read from .flutter-plugin-dependencies instead. The contents are broken down by platform:

Example:
```
{
    "info": "This is a generated file; do not edit or check into version control.",
    "plugins": {
        "ios": [
            {
                "name": "connectivity",
                "path": "C:\\\\Users\\\\Jonah\\\\AppData\\\\Roaming\\\\Pub\\\\Cache\\\\hosted\\\\pub.dartlang.org\\\\connectivity-0.4.8+2\\\\",
                "dependencies": []
            },
            {
                "name": "device_info",
                "path": "C:\\\\Users\\\\Jonah\\\\AppData\\\\Roaming\\\\Pub\\\\Cache\\\\hosted\\\\pub.dartlang.org\\\\device_info-0.4.2+1\\\\",
                "dependencies": []
            },
            {
                "name": "url_launcher",
                "path": "C:\\\\Users\\\\Jonah\\\\AppData\\\\Roaming\\\\Pub\\\\Cache\\\\hosted\\\\pub.dartlang.org\\\\url_launcher-5.4.2\\\\",
                "dependencies": []
            },
            {
                "name": "video_player",
                "path": "C:\\\\Users\\\\Jonah\\\\AppData\\\\Roaming\\\\Pub\\\\Cache\\\\hosted\\\\pub.dartlang.org\\\\video_player-0.10.6\\\\",
                "dependencies": []
            }
        ],
        "android": [
            {
                "name": "connectivity",
                "path": "C:\\\\Users\\\\Jonah\\\\AppData\\\\Roaming\\\\Pub\\\\Cache\\\\hosted\\\\pub.dartlang.org\\\\connectivity-0.4.8+2\\\\",
                "dependencies": []
            },
            {
                "name": "device_info",
                "path": "C:\\\\Users\\\\Jonah\\\\AppData\\\\Roaming\\\\Pub\\\\Cache\\\\hosted\\\\pub.dartlang.org\\\\device_info-0.4.2+1\\\\",
                "dependencies": []
            },
            {
                "name": "url_launcher",
                "path": "C:\\\\Users\\\\Jonah\\\\AppData\\\\Roaming\\\\Pub\\\\Cache\\\\hosted\\\\pub.dartlang.org\\\\url_launcher-5.4.2\\\\",
                "dependencies": []
            },
            {
                "name": "video_player",
                "path": "C:\\\\Users\\\\Jonah\\\\AppData\\\\Roaming\\\\Pub\\\\Cache\\\\hosted\\\\pub.dartlang.org\\\\video_player-0.10.6\\\\",
                "dependencies": []
            }
        ],
```

I thought we filled an issue for this, maybe @franciscojma86 knows where it is 😄

**Comment by franciscojma86** [comment]
I thought I has a draft PR for that but can't find it :( But yeah, `flutter-plugins` should/will be deprecated anyways, so you should use `.flutter-plugins-dependencies`. 

The only issue that I know of is https://github.com/flutter/flutter/issues/39657 but it's already referenced here.

**Comment by blasten** [comment]
ahh good call

File: packages/flutter_tools/templates/module/android/library/include_flutter.groovy.copy.tmpl [hunk 1]
```diff
@@ -1,37 +1,17 @@
...
-gradle.include ':flutter'
-gradle.project(':flutter').projectDir = new File(flutterProjectRoot, '.android/Flutter')
+def properties = new Properties()
 
-def plugins = new Properties()
-def pluginsFile = new File(flutterProjectRoot, '.flutter-plugins')
-if (pluginsFile.exists()) {
-    pluginsFile.withReader('UTF-8') { reader -> plugins.load(reader) }
-}
+assert localPropertiesFile.exists(), "❗️The Flutter module doesn't have a `$localPropertiesFile` file." +
```
Comment (blasten) [inline packages/flutter_tools/templates/module/android/library/include_flutter.groovy.copy.tmpl:11]: FYI @xster. This new error message will be useful when building the host app without running `pub get` in the module.

**Comment by blasten** [comment]
I've moved most of the logic in `settings.gradle` to the Flutter SDK location, so if we have to make a change later on, it's easier to manage. PTAL

File: packages/flutter_tools/templates/module/android/library/include_flutter.groovy.copy.tmpl (hunk 1 above)
Comment (xster) [reply packages/flutter_tools/templates/module/android/library/include_flutter.groovy.copy.tmpl:14]: Awesome! Thanks. 

nit: trailing period.

Comment (blasten) [reply packages/flutter_tools/templates/module/android/library/include_flutter.groovy.copy.tmpl:14]: Done

**Review by jonahwilliams — APPROVED** [review]
Overal LGTM

Do we plan to handle migrating users too? If not, we might want to keep a project using the old settings.gradle for testing

**Comment by blasten** [comment]
Yep - it will be migrated. I filed https://github.com/flutter/flutter/issues/54566 and added an integration test just in case, so we don't delete `.flutter-plugins` prematurely.

**Pull request merged**